### PR TITLE
fix: add `get_column` function for Query obj

### PR DESCRIPTION
**Pull request by hughhhh**
### SUMMARY

With [this PR](https://github.com/apache/superset/pull/21378), we are now using the get_column function within the BaseDatasource, but we didn't incorporate this inside Query model from the original spec. Adding it now to fix the issue for queries being able to execute inside explore

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset/models/sql_lab.py [hunk 1]
```diff
@@ -183,7 +182,7 @@ def sql_tables(self) -> List[Table]:
         return list(ParsedQuery(self.sql).tables)
 
     @property
-    def columns(self) -> List[ResultSetColumnType]:
+    def columns(self) -> List[Dict[str, Any]]:
```
Comment (Antonio-RiveroMartnez) [inline superset/models/sql_lab.py:185]: Not for changing, just curious why moving from a typed dict to a general one?

**Review by Antonio-RiveroMartnez — APPROVED** [review]
LGTM!

**Review by pkdotson — APPROVED** [review]
lgtm!

**Review by AAfghahi — APPROVED** [review]

**Comment by jinghua-qa** [comment]
I found an issue when i tested in the ephemeral env:
when i make a query in sql, the column is date but when i create chart and open in explore, the date column become string
 repo steps:
1, query : Select * From "birth_names"
2, observe the column type of ds is "Datetime"
3, create chart and open the query in explore
4, observe the ds column type

Expected:
same column type in sql and explore

Actual:
datetime column become string in explore, user can not create time series related charts


https://user-images.githubusercontent.com/81597121/193900058-fb600d05-73f3-45d7-9266-38274842f517.mov

**Comment by sadpandajoe** [comment]
🏷️ preset:2022.39

**Pull request merged**